### PR TITLE
Implementation of the nonlinears.

### PR DESCRIPTION
**Pull request by ufucuk**
I've implemented nonlinears. I'll also write test cases soon, but they are working fine at the moment.

### CODE REVIEW

File: src/creative.coffee [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
+  constructor: ->
+    super
+    @type = "linear"
+    @duration = 0
+    @skipDelay = null
+    @mediaFiles = []
+    @videoClickThroughURLTemplate = null
+    @videoClickTrackingURLTemplates = []
+    @videoCustomClickURLTemplates = []
+    @adParameters = null
```
Comment (rs) [inline src/creative.coffee]: Please keep project’s style in term of indentation.

**Comment by rs** [comment]
LGTM, please fix indentation and I’ll merge.

**Comment by ufucuk** [comment]
Indentation is fixed!

**Comment by rs** [comment]
Last thing, could you please squash your two commits?

**Comment by ufucuk** [comment]
Squashed.

**Comment by rs** [comment]
Now we have 4 commits, not quite squashed :)

https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request#squash-your-changes

**Pull request closed without merging**